modules/mig/variables.tf

```diff
@@ -50,7 +50,7 @@ variable "distribution_policy_zones" {
 
 variable "update_policy" {
   description = "The rolling update policy. https://www.terraform.io/docs/providers/google/r/compute_region_instance_group_manager.html#rolling_update_policy"
-  type        = "list"
+  type        = list(map(string))
```

Comment:
This should be typed as `list(map(any))` as some of the arguments are numbers.

Reply:
In the current case, the type of the first variable of this map will be number, so all other values will ben assigned as the numbers too...Better is:
variable "update_policy" {
  description = "The rolling update policy. https://www.terraform.io/docs/providers/google/r/compute_region_instance_group_manager.html#rolling_update_policy"
  type        = list(object({
    max_surge_fixed         = number
    max_surge_percent       = number
    max_unavailable_fixed   = number
    max_unavailable_percent = number
    min_ready_sec           = number
    minimal_action          = string
    type                    = string
  }))
  default     = []
}
I've tested it, it works

Reply:
Map should not be used when values are of different types